shared/logic/IntraTickCache.ts

```diff
@@ -51,6 +51,18 @@ export function getBuildingIO(
             resources[k] = 1;
          });
       }
+      if ("sellResources" in b && type === "output") {
+         const capacity = b.capacity * b.level * totalMultiplierFor(xy, "output", 1, gs);
+         forEach((b as IMarketBuildingData).sellResources, (k) => {
+            const buyResource = (b as IMarketBuildingData).availableResources[k]!;
+            result[buyResource] = round(
+               (capacity * getMarketPrice(k, xy, gs)) / getMarketPrice(buyResource, xy, gs),
+               1,
+            );
+         });
+         // Market output is not affected by multipliers
+         return result;
```

Comment:
@insraq Should both of these set the cache before returning?

_cache.buildingIO.set(key, Object.freeze(result));

Reply:
Yes, cache should be set before returning :-)
It's not a big deal (potentially doing more calculations) but we should fix this just for consistency's sake.

Reply:
Cache is now set on all returns